src/mongocxx/collection.cpp

```diff
@@ -624,9 +624,15 @@ bsoncxx::document::value collection::create_index(bsoncxx::document::view keys,
         throw exception::operation(std::make_tuple(error.message, error.code));
     }
 
-    // TODO: return the response from the server, this is not possible now due to the way
-    // libmongoc works.
-    return bsoncxx::document::value{bsoncxx::document::view{}};
+    if (options.name()) {
+        return bsoncxx::builder::stream::document{} 
```

Comment:
I think this could be a bit cleaner if you use the [value_or](http://mnmlstc.github.io/core/optional.html#optional:T:::value_or__URR) method on optional. It could look something like this:
```
bsoncxx::builder::stream::document{};
document << "name" << options.name.value_or(std::string{libmongoc::collection_keys_to_index_string(bson_keys.bson()));
return document.extract();